plonetheme/onegovbear/browser/forms.py

```diff
@@ -0,0 +1,173 @@
+    footer_background_color = Variable('$footer-background-color')
+    footer_font_color = Variable('$footer-font-color')
+    footer_link_color = Variable('$footer-link-color')
+    footer_link_color_hover = Variable('$footer-link-color-hover')
+    font_family_base = Variable('$font-family-base')
+    line_height_base = Variable('$line-height-base')
+    font_size_base = Variable('$font-size-base')
+    font_size_h1 = Variable('$font-size-h1')
+    font_size_h2 = Variable('$font-size-h2')
+    font_size_h3 = Variable('$font-size-h3')
+    page_bg_color = Variable('$page-bg-color')
+    gray_base = Variable('$gray-base')
+    gray_darker = Variable('$gray-darker')
+    gray_dark = Variable('$gray-dark')
+    gray = Variable('$gray')
+    gray_light = Variable('$gray-light')
+    gray_lighter = Variable('$gray-lighter')
+
+
+class VariablesConfig(object):
+    """
+    This proxy object saves the name of the form field, its value and
+    the SCSS variable name. We need the SCSS variable name later when
+    returning the variables in the dynamic SCSS resource factory.
+    Example:
+
+        {'service_nav_link_color': {
+            'value': u'green',
+            'variable_name': '$service-nav-link-color'}
+        }
+    """
+    implements(IVariablesSchema)
+    _protected_names = ['storage', 'fields']
+
+    def __init__(self, storage, fields):
+        self.storage = storage
+        self.fields = fields
+
+    def __getattr__(self, name):
+        if name in self._protected_names:
+            return object.__getattr__(self, name)
+        value = self.storage.get(name)
+        return value.get('value')
+
+    def __setattr__(self, name, value):
+        if name in self._protected_names:
+            return object.__setattr__(self, name, value)
+        if value:
+            self.storage[name] = {
+                'value': value,
+                'variable_name': self.fields._data[name].field.variable_name
+            }
+        elif name in self.storage:
+            del self.storage[name]
+
+
+class VariablesForm(form.SchemaEditForm):
+    """
+    This form is used to customize the SCSS variables. Subclasses must
+    override `annotation_key` and `config` and implement their own config
+    class with a custom schema.
+    """
+    label = _(u'variables_form_label', default=u'Custom SCSS variables')
+    schema = IVariablesSchema
+    ignoreContext = False
+    annotation_key = ANNOTATION_KEY
+    config = VariablesConfig
+
+    @property
+    def description(self):
+        description = translate(
+            _(u'variables_form_description',
+              default=u'The values entered in this form will override '
+                      u'the SCSS variables defined in the theme for the '
+                      u'INavigationRoot (e.g. Plone Site and Subsites).'),
+            context=self.request)
+
+        # Add a special note if we're on a Subsite.
+        if ISubsite and ISubsite.providedBy(self.context):
```

Comment:
Actually not only subsites by ftw.subsites are behaving like this, but all navigation roots which are not the site root, right? So you could use `INavigationRoot` here and drop the dependency to ftw.subsite :wink: